consensus/enclave/impl/src/lib.rs

```diff
@@ -363,24 +365,57 @@ impl SgxConsensusEnclave {
     }
 
     /// Validate a list of MintTxs.
-    fn validate_mint_txs(&self, mint_txs: &[MintTx]) -> Result<()> {
-        // TODO: iterate over each indivdual transaction and validate it. This requires
-        // the enclave has knowledge of active minting configurations.
-        // This validation already happens in untruested, but it will be nice to do it
-        // here as well. This will get implemetend in a follow up PR.
-
-        // Ensure all nonces are unique.
+    fn validate_mint_txs(
+        &self,
+        mint_txs_with_config: Vec<(MintTx, MintConfigTx, MintConfig)>,
+        current_block_index: u64,
+        config: &BlockchainConfig,
+    ) -> Result<Vec<MintTx>> {
+        // Ensure all nonces are unique and all transactions matches the associated
+        // configuration.
+        let mut mint_txs = Vec::with_capacity(mint_txs_with_config.len());
         let mut seen_nonces = BTreeSet::default();
-        for tx in mint_txs {
-            if !seen_nonces.insert(tx.prefix.nonce.clone()) {
+        for (mint_tx, mint_config_tx, mint_config) in mint_txs_with_config {
+            // The nonce should be unique.
+            if !seen_nonces.insert(mint_tx.prefix.nonce.clone()) {
                 return Err(Error::FormBlock(format!(
                     "Duplicate MintTx nonce: {:?}",
-                    tx.prefix.nonce
+                    mint_tx.prefix.nonce
+                )));
+            }
+
+            // The associated MintConfigTx should be for the same token being minted.
+            if mint_config_tx.prefix.token_id != mint_tx.prefix.token_id {
+                return Err(Error::FormBlock(format!(
+                    "MintTx token_id does not match MintConfigTx token_id: {:?} != {:?}",
+                    mint_tx.prefix.token_id, mint_config_tx.prefix.token_id
+                )));
+            }
+
+            // The specific MintConfig used should be part of the provided MintConfigTx.
+            if !mint_config_tx.prefix.configs.contains(&mint_config) {
+                return Err(Error::FormBlock(format!(
+                    "MintTx {} referenced a MintConfig that is not part of the MintConfigTx",
+                    mint_tx,
                 )));
             }
+
+            // The associated MintConfigTx should be valid.
+            self.validate_mint_config_txs(vec![mint_config_tx], current_block_index, config)?;
+
+            // The MintTx should be valid.
+            validate_mint_tx(
+                &mint_tx,
+                current_block_index,
+                config.block_version,
+                &mint_config,
+            )?;
```

Comment:
This is all the extra validation that we are now able to do, once we have the `MintTx` coupled with the configuration data.